org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/preferences/MavenPreferenceConstants.java

```diff
@@ -166,6 +166,13 @@
    **/
   String P_AUTO_UPDATE_CONFIGURATION = PREFIX + "autoUpdateProjects"; //$NON-NLS-1$
 
+  /**
+   * boolean
+   *
+   * @since 2.0
+   **/
+  String P_ALWAYS_CONFIGURE_DEFAULT_JRE = PREFIX + "alwaysConfigureDefaultJRE"; //$NON-NLS-1$
+
```

Comment:
Those preferences that purely belong to JDT integration should be part of the org.eclipse.m2e.jdt bundle

Reply:
So a new preferences page just for this boolean flag?

Reply:
Yes. And maybe other will follow.
Also, please define in the dialog what you mean by the "default JRE". Default is not an explicit enough terms as Maven and JDT have their defaults. If this is JDT's default JRE, the preference page could link to JDT's one.